src/containers/tags/withTags.js

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { inject } from "mobx-react";
+import { Query } from "react-apollo";
+import tagsQuery from "./tags.gql";
+
+/**
+ * withTag higher order query component for fetching tag information
+ * @name WithTag
+ * @param {React.Component} Component to decorate
+ * @returns {React.Component} - Component with `tag` prop
+ */
+export default (Component) => (
+  @inject("primaryShopId")
+  class WithTag extends React.Component {
+    static propTypes = {
+      /**
+       * ShopId used to obtain tags for
+       */
+      primaryShopId: PropTypes.string.isRequired
+    }
+
+    render() {
+      const { primaryShopId } = this.props;
+
+      return (
+        <Query query={tagsQuery} variables={{ shopId: primaryShopId }}>
```

Comment:
It might be worth merging some of the functionality of `withNavigationTags` into this HOC to fetch all tags.

`withNavigationTags` will continue to fetch tags until all have been loaded, this is necessary for now due to the hard limit on each request, which should be about 200,

Reply:
Good idea.

Tags are now fetched inside of `withTags`, and `withNavigationTags` uses this data instead of running a separate query.